Scripts/Runtime/Jobs/Deferred/IJobDeferredNativeArrayForBatch.cs

```diff
@@ -0,0 +1,140 @@
+                                                                      atomicSafetyHandlePtr);
+
+            return dependsOn;
+        }
+
+
+        internal struct JobDeferredNativeArrayForBatchProducer<TJob>
+            where TJob : struct, IJobDeferredNativeArrayForBatch
+        {
+            // ReSharper disable once StaticMemberInGenericType
+            internal static readonly SharedStatic<IntPtr> s_JobReflectionData = SharedStatic<IntPtr>.GetOrCreate<JobDeferredNativeArrayForBatchProducer<TJob>>();
+
+            [Preserve]
+            internal static void Initialize()
+            {
+                if (s_JobReflectionData.Data == IntPtr.Zero)
+                {
+#if UNITY_2020_2_OR_NEWER
+                    s_JobReflectionData.Data = JobsUtility.CreateJobReflectionData(typeof(TJob),
+                                                                                   typeof(TJob),
+                                                                                   (ExecuteJobFunction)Execute);
+#else
+                    s_JobReflectionData.Data = JobsUtility.CreateJobReflectionData(typeof(TJob),
+                                                                              typeof(TJob),
+                                                                              JobType.ParallelFor,
+                                                                              (ExecuteJobFunction)Execute);
+#endif
+                }
+            }
+
+            private delegate void ExecuteJobFunction(ref TJob jobData,
+                                                     IntPtr additionalPtr,
+                                                     IntPtr bufferRangePatchData,
+                                                     ref JobRanges ranges,
+                                                     int jobIndex);
+
+
+            [SuppressMessage("ReSharper", "MemberCanBePrivate.Global", Justification = "Required by Burst.")]
+            public static unsafe void Execute(ref TJob jobData,
+                                              IntPtr additionalPtr,
+                                              IntPtr bufferRangePatchData,
+                                              ref JobRanges ranges,
+                                              int jobIndex)
+            {
+                while (true)
+                {
+                    if (!JobsUtility.GetWorkStealingRange(ref ranges, jobIndex, out int beginIndex, out int endIndex))
+                    {
+                        return;
+                    }
+
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+                    JobsUtility.PatchBufferMinMaxRanges(bufferRangePatchData, UnsafeUtility.AddressOf(ref jobData), beginIndex, endIndex - beginIndex);
+#endif
+
+                    jobData.Execute(beginIndex, endIndex - beginIndex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method is only to be called by automatically generated setup code.
```

Comment:
```suggestion
        /// This method is only to be called by automatically generated setup code.
        /// Called by Unity automatically.
```
Should we add a `<see cref="" /> to a Unity implemented place this is done in case it is removed/breaks/changes in the future?